src/main/scala/beam/agentsim/agents/ridehail/allocation/PoolingAlonsoMora.scala

```diff
@@ -0,0 +1,174 @@
+
+  override def respondToInquiry(inquiry: RideHailRequest): InquiryResponse = {
+    rideHailManager.vehicleManager
+      .getClosestIdleVehiclesWithinRadiusByETA(
+        inquiry.pickUpLocationUTM,
+        rideHailManager.radiusInMeters,
+        inquiry.departAt
+      )
+      .headOption match {
+      case Some(agentETA) =>
+        SingleOccupantQuoteAndPoolingInfo(agentETA.agentLocation, Some(PoolingInfo(1.1, 0.6)))
+      case None =>
+        NoVehiclesAvailable
+    }
+  }
+
+
+  override def allocateVehiclesToCustomers(
+                                            tick: Int,
+                                            vehicleAllocationRequest: AllocationRequests
+                                          ): AllocationResponse = {
+    logger.debug("buffer size: {}",vehicleAllocationRequest.requests.size)
+    var toPool: Set[RideHailRequest] = Set()
+    var notToPool: Set[RideHailRequest] = Set()
+    var allocResponses: List[VehicleAllocation] = List()
+    var alreadyAllocated: Set[Id[Vehicle]] = Set()
+    vehicleAllocationRequest.requests.foreach {
+      case (request, routingResponses) if routingResponses.isEmpty =>
+        toPool += request
+      case (request, _) =>
+        notToPool += request
+    }
+    notToPool.foreach { request =>
+      val routeResponses = vehicleAllocationRequest.requests(request)
+      val indexedResponses = routeResponses.map(resp => (resp.requestId -> resp)).toMap
+
+
+      // First check for broken route responses (failed routing attempt)
+      if (routeResponses.find(_.itineraries.size == 0).isDefined) {
+        allocResponses = allocResponses :+ NoVehicleAllocated(request)
+      } else {
+        // Make sure vehicle still available
+        val vehicleId = routeResponses.head.itineraries.head.legs.head.beamVehicleId
+        if (rideHailManager.vehicleManager.getIdleVehicles.contains(vehicleId) && !alreadyAllocated.contains(vehicleId)) {
+          alreadyAllocated = alreadyAllocated + vehicleId
+          val mobilityServiceRequests = tempScheduleStore.remove(request.requestId).get
+
+          val pickDropIdAndLegs = mobilityServiceRequests.map{ req =>
+            req.routingRequestId match {
+              case Some(routingRequestId) =>
+                PickDropIdAndLeg(req.person.get,indexedResponses(routingRequestId).itineraries.head.legs.headOption)
+              case None =>
+                PickDropIdAndLeg(req.person.get,None)
+            }
+          }
+          allocResponses = allocResponses :+ VehicleMatchedToCustomers(
+            request,
+            rideHailManager.vehicleManager.getIdleVehicles(vehicleId),
+            pickDropIdAndLegs
+          )
+        } else {
+          allocResponses = allocResponses :+ NoVehicleAllocated(request)
+          request.groupedWithOtherRequests.foreach { req =>
+            allocResponses = allocResponses :+ NoVehicleAllocated(req)
+          }
+        }
+      }
+    }
+    if(toPool.size > 0){
```

Comment:
Would it be better to just use toPool.map, then return a tuple2 of the customerReqs and IdToReqs, then map off of that? It might even be more efficient as it would loop once instead of twice